rust/src/encodings/dictionary.rs

```diff
@@ -82,14 +82,14 @@ impl<'a> Decoder for DictionaryDecoder<'a> {
             assert!(index_type.as_ref().is_dictionary_key_type());
 
             match index_type.as_ref() {
-                Int8 => self.decode_index::<Int8Type>().await,
-                Int16 => self.decode_index::<Int16Type>().await,
-                Int32 => self.decode_index::<Int32Type>().await,
-                Int64 => self.decode_index::<Int64Type>().await,
-                UInt8 => self.decode_index::<UInt8Type>().await,
-                UInt16 => self.decode_index::<UInt16Type>().await,
-                UInt32 => self.decode_index::<UInt32Type>().await,
-                UInt64 => self.decode_index::<UInt64Type>().await,
+                Int8 => self.decode_index::<Int8Type>(index_type).await,
```

Comment:
Can this directly use `self.decode_index(index_type)` so that we dont need the pattern matching arms